lib/mayaUsd/fileio/utils/meshWriteUtils.cpp

```diff
@@ -477,4 +478,76 @@ UsdMayaMeshUtil::assignSubDivTagsToUSDPrim(MFnMesh& meshFn,
     }
 }
 
+void
+UsdMayaMeshUtil::writeVertexData(const MFnMesh& meshFn,
+                                 UsdGeomMesh& primSchema,
+                                 const UsdTimeCode& usdTime,
+                                 UsdUtilsSparseValueWriter& valueWriter)
+{
+    MStatus status{MS::kSuccess};
+
+    const uint32_t numVertices = meshFn.numVertices();
+    VtArray<GfVec3f> points(numVertices);
+    const float* pointsData = meshFn.getRawPoints(&status);
+    if(status)
+    {
+        // use memcpy() to copy the data. HS April 09, 2020
+        memcpy((GfVec3f*)points.data(), pointsData, sizeof(float) * 3 * numVertices);
+
+        VtArray<GfVec3f> extent(2);
+        // Compute the extent using the raw points
+        UsdGeomPointBased::ComputeExtent(points, &extent);
+
+        UsdMayaWriteUtil::SetAttribute(primSchema.GetPointsAttr(), &points, valueWriter, usdTime);
+        UsdMayaWriteUtil::SetAttribute(primSchema.CreateExtentAttr(), &extent, valueWriter,usdTime);
+    }
+    else
+    {
+        MGlobal::displayError(MString("Unable to access mesh vertices on mesh: ") + meshFn.fullPathName());
+    }
+}
+
+void 
+UsdMayaMeshUtil::writeFaceVertexIndicesData(const MFnMesh& meshFn, 
+                                            UsdGeomMesh& primSchema, 
+                                            const UsdTimeCode& usdTime, 
+                                            UsdUtilsSparseValueWriter& valueWriter)
+{
+    const int numFaceVertices = meshFn.numFaceVertices();
+    const int numPolygons = meshFn.numPolygons();
+
+    VtArray<int> faceVertexCounts(numPolygons);
+    VtArray<int> faceVertexIndices(numFaceVertices);
+    MIntArray mayaFaceVertexIndices; // used in loop below
+    unsigned int curFaceVertexIndex = 0;
+    for (int i = 0; i < numPolygons; i++) {
+        meshFn.getPolygonVertices(i, mayaFaceVertexIndices);
+        faceVertexCounts[i] = mayaFaceVertexIndices.length();
+        for (unsigned int j=0; j < mayaFaceVertexIndices.length(); j++) {
+            faceVertexIndices[ curFaceVertexIndex ] = mayaFaceVertexIndices[j]; // push_back
+            curFaceVertexIndex++;
+        }
+    }
+    UsdMayaWriteUtil::SetAttribute(primSchema.GetFaceVertexCountsAttr(), &faceVertexCounts, valueWriter, usdTime);
+    UsdMayaWriteUtil::SetAttribute(primSchema.GetFaceVertexIndicesAttr(), &faceVertexIndices, valueWriter, usdTime);
+}
+
+void 
+UsdMayaMeshUtil::writeInvisibleFacesData(const MFnMesh& meshFn, 
+                                         UsdGeomMesh& primSchema, 
+                                         UsdUtilsSparseValueWriter& valueWriter)
+{
+    MUintArray mayaHoles = meshFn.getInvisibleFaces();
+    const uint32_t count = mayaHoles.length();
```

Comment:
Just `unsigned int` here?

Reply:
I think uint32_t should be Ok here. Any particular concern why this needs to change here?

Reply:
Same argument as above.